### DISCUSSION TITLE
plugin load failed

### DISCUSSION THREAD

**Discussion by Eden7Ba23** [YunYouJun/hexo-theme-starry#252] [Q&A 疑难杂症] [answered]
上次更新时，最新的Yun主题是1.7.0版本。
本次更新的时候，看到了1.8.0的迁移，所以应用了以下更改
https://github.com/Eden7Ba23/BlogBackup/commit/f618fffee4e5056541eaba110428e7e02806bf34
这应该不出问题，但是实际上，No Layout问题出现了，导致生成的文件全部为空
我本以为是npm依赖问题，但是在我电脑本地，问题依然出现。
问题貌似出在了：
```
ERROR Plugin load failed: hexo-theme-yun
Error: EISDIR: illegal operation on a directory, read
```
这个问题该怎么解决呢？

**Comment by YunYouJun** [comment, marked as answer]
迁移指南见：https://yun.yunyoujun.cn/guide/migrate.html

实际上你先在本地博客目录 `npm i hexo-theme-yun` 一下，然后它会锁到你的 package.json。
然后你那几句话就都可以删掉了。

可以参见我的 [gh-pages.yml](https://github.com/YunYouJun/yunyoujun.github.io/blob/hexo/.github/workflows/gh-pages.yml).

**Reply by Eden7Ba23** [reply]
看起来没有效果。明显在运行过程中新的Package.json已经生成，问题出在是个目录上。

**Reply by Eden7Ba23** [reply]
我后面还尝试过在不同的机器上面clone然后 npm -i 但都是 EISDIR

**Reply by YunYouJun** [reply]
你不需要写这些：

![image](https://user-images.githubusercontent.com/25154432/152986783-6785c042-ca58-46f5-a7c6-f750725242c7.png)

也不需要在 actions 的时候再去按照 hexo-theme-yun，你应该上传的时候就把依赖包括进去，然后只需要执行 `npm i`。

你可以 clone 我的仓库，然后 npm i 试试。

**Reply by Eden7Ba23** [reply]
> 你不需要写这些：
> 
> ![image](https://user-images.githubusercontent.com/25154432/152986783-6785c042-ca58-46f5-a7c6-f750725242c7.png)
> 
> 也不需要在 actions 的时候再去按照 hexo-theme-yun，你应该上传的时候就把依赖包括进去，然后只需要执行 `npm i`。
> 
> 你可以 clone 我的仓库，然后 npm i 试试。

等一下，咱这边不是更新过了吗？可以确定package.json也有yun的依赖。

https://github.com/Eden7Ba23/BlogBackup/commit/37c54445ceca0f803b62ff1d4251d6cba88da6a9

**Reply by YunYouJun** [reply]
不需要这里：

![image](https://user-images.githubusercontent.com/25154432/153183753-e1b781c4-e60b-4df7-9ecf-2a6c31b5bea4.png)

**Reply by Eden7Ba23** [reply]
> 不需要这里：
> 
> ![image](https://user-images.githubusercontent.com/25154432/153183753-e1b781c4-e60b-4df7-9ecf-2a6c31b5bea4.png)

Not Work.Tested on https://github.com/Eden7Ba23/BlogBackup/commit/3ac36df963798593f45fd0196732089ac3206c75 build with https://github.com/Eden7Ba23/BlogBackup/commit/edc60044a5eb25f1b524187cd5575c4d6ea6e413 .

应该明确一下问题。出在加载yun的时候的：
ERROR Plugin load failed: hexo-theme-yun
Error: EISDIR: illegal operation on a directory, read

**Reply by YunYouJun** [reply]
破案了，hexo 版本太低（4.2.1），没升级吧。升级到 6.0。

**Reply by Eden7Ba23** [reply]
> 

Done.
